src/main/java/ch/njol/skript/expressions/ExprPlayerViewDistance.java

```diff
@@ -0,0 +1,118 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+package ch.njol.skript.expressions;
+
+import org.bukkit.Bukkit;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.util.Getter;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+
+@Name("Player View Distance")
+@Description("The view distance of a player. This expression requires PaperMC.")
+@Examples({"set view distance of player to 10", "set {_view} to view distance of player",
+		"reset view distance of all players", "add 2 to view distance of player"})
+@Since("INSERT VERSION")
+public class ExprPlayerViewDistance extends PropertyExpression<Player, Integer> {
+	
+	static {
+		if (Skript.methodExists(Player.class, "getViewDistance"))
+			register(ExprPlayerViewDistance.class, Integer.class, "view distance", "players");
+	}
+	
+	@SuppressWarnings({"unchecked", "null"})
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
+		setExpr((Expression<Player>) exprs[0]);
+		return true;
+	}
+	
+	@Override
+	protected Integer[] get(Event e, Player[] source) {
+		return get(source, new Getter<Integer, Player>() {
+			@SuppressWarnings("null")
+			@Override
+			public Integer get(Player arg) {
+				return arg.getViewDistance();
+			}
+		});
+	}
+	
+	@Nullable
+	@Override
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		if (mode == ChangeMode.REMOVE_ALL)
+			return null;
+		return CollectionUtils.array(Number.class);
+	}
+	
+	@Override
+	public void change(Event e, @Nullable Object[] delta, ChangeMode mode) {
+		int distance = delta == null ? 0 : ((Number) delta[0]).intValue();
+		switch (mode) {
+			case DELETE:
+			case SET:
+				for (Player player : getExpr().getArray(e)) {
+					player.setViewDistance(distance);
```

Comment:
The value to be changed to can be stored in a variable and then can be changed using only one for loop at the end

Reply:
kinda hard to do if you are adding/subtracting from all players and all players may or may not have different view distances set already

Reply:
ugh you are right